### ISSUE TITLE
Rhodes Example: TypeError: can't convert expression to float

### ISSUE THREAD

**Issue by ntonnaett** [pyphs/pyphs#129]
As mentioned in PR #128 I'm trying to fix the rhodes example. With the develop branch I'm getting this error now.

Traceback (most recent call last):
  File "rhodes_new.py", line 95, in <module>
    inits={'x': x0})
  File "/home/nils/Development/pyphs/pyphs/numerics/numerical_method/_method.py", line 311, in to_simulation
    return Simulation(self, inits=inits, config=config, erase=erase)
  File "/home/nils/Development/pyphs/pyphs/numerics/simulations/simulation.py", line 182, in __init__
    self.init_numericalCore(erase=erase)
  File "/home/nils/Development/pyphs/pyphs/numerics/simulations/simulation.py", line 250, in init_numericalCore
    config=self.config_numeric())
  File "/home/nils/Development/pyphs/pyphs/numerics/cpp/method2cpp.py", line 233, in method2cpp
    parameters_files = parameters(subs, objlabel)
  File "/home/nils/Development/pyphs/pyphs/numerics/cpp/method2cpp.py", line 289, in parameters
    _append_subs(subs, files)
  File "/home/nils/Development/pyphs/pyphs/numerics/cpp/method2cpp.py", line 331, in _append_subs
    files['cpp'] += str(float(s[k])) + ', '
  File "/home/nils/.local/lib/python3.7/site-packages/sympy/core/expr.py", line 280, in __float__
    raise TypeError("can't convert expression to float")
TypeError: can't convert expression to float

**Comment by afalaize** [comment]
Hello @ntonnaett! Thank you for contibuting!

Yes, we are in a big mess here. The error occurs at c++ code generation, more preciely at generation of the array associated with the parameters contained in `method.subs` inherited from `core.subs` in
`File "/home/nils/Development/pyphs/pyphs/numerics/cpp/method2cpp.py", line 331, in _append_subs`

The problem is that the parameter with symbol `Pickupf` is an expression, not a numerical value:

```
>>> core.subs[core.symbols('Pickupf')]
1/(2*Pickup_Rb**2*Pickup_Rp*dtq_PickupOBS*((-Pickup_Lh**2 + (Pickup_Lv - Pickup_Rp + q_PickupOBS)**2)/(Pickup_Lh**2 + (Pickup_Lv - Pickup_Rp + q_PickupOBS)**2)**2 - (-Pickup_Lh**2 + (Pickup_Lv + Pickup_Rp + q_PickupOBS)**2)/(Pickup_Lh**2 + (Pickup_Lv + Pickup_Rp + q_PickupOBS)**2)**2))
```

 At this point, it should have been substituted everywhere in `method` before entering the cpp code generation module, with a call to the `core` method `substitute(selfexprs=True)` (from the `core` or the `method`). At least, this was the case in pyphs 0.3 (current master which is broken due to dependencies).

I'll have a look as soon as possible

**Comment by afalaize** [comment]
solved by PR #130
